security/cmd/chiron/main.go

```diff
@@ -0,0 +1,209 @@
+	// The length of certificate rotation grace period, configured as the ratio of the certificate TTL.
+	// If certGracePeriodRatio is 0.2, and cert TTL is 24 hours, then the rotation will happen
+	// after 24*(1-0.2) hours since the cert is issued.
+	certGracePeriodRatio float32
+	// The minimum grace period for cert rotation.
+	certMinGracePeriod time.Duration
+
+	// Whether to generate PKCS#8 private keys.
+	pkcs8Keys bool
+
+	// Monitoring port number
+	monitoringPort int
+	// Enable profiling in monitoring
+	enableProfiling bool
+
+	logOptions *log.Options
+	// Currently, no topic is registered for ctrlz yet
+	ctrlzOptions *ctrlz.Options
+
+	// Whether enable the webhook controller
+	enableController bool
+
+	// The name of the MutatingWebhookConfiguration to manage
+	mutatingWebhookConfigName string
+	mutatingWebhookName       string
+
+	// TODO (lei-tang): Add the name of the ValidatingWebhookConfiguration to manage
+}
+
+func fatalf(template string, args ...interface{}) {
+	if len(args) > 0 {
+		log.Errorf(template, args...)
+	} else {
+		log.Errorf(template)
+	}
+	os.Exit(-1)
+}
+
+func init() {
+	flags := rootCmd.Flags()
+
+	flags.BoolVar(&opts.enableController, "enable-controller", false, "Specifies whether enabling "+
+		"Istio Webhook Controller.")
+	flags.StringVar(&opts.certificateNamespace, "certificate-namespace", "istio-system",
+		"The namespace of the webhook certificates.")
+
+	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
+		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
+
+	// Monitoring configuration
+	flags.IntVar(&opts.monitoringPort, "monitoring-port", 15021, "The port number for monitoring Chiron. "+
+		"If unspecified, Chiron will disable monitoring.")
+	flags.BoolVar(&opts.enableProfiling, "enable-profiling", false, "Enabling profiling when monitoring Chiron.")
+
+	// Certificate issuance configuration.
+	flags.DurationVar(&opts.certTTL, "cert-ttl", cmd.DefaultWorkloadCertTTL,
+		"The TTL of issued certificates.")
+	flags.DurationVar(&opts.maxCertTTL, "max-cert-ttl", cmd.DefaultMaxWorkloadCertTTL,
+		"The max TTL of issued certificates.")
+	flags.Float32Var(&opts.certGracePeriodRatio, "cert-grace-period-ratio",
+		cmd.DefaultWorkloadCertGracePeriodRatio, "The certificate rotation grace period, as a ratio of the "+
+			"certificate TTL.")
+	flags.DurationVar(&opts.certMinGracePeriod, "cert-min-grace-period",
+		cmd.DefaultWorkloadMinCertGracePeriod, "The minimum certificate rotation grace period.")
+	flags.BoolVar(&opts.pkcs8Keys, "pkcs8-keys", false, "Whether to generate PKCS#8 private keys.")
+
+	// MutatingWebhook configuration
+	flags.StringVar(&opts.mutatingWebhookConfigName, "mutating-webhook-config-name", "istio-sidecar-injector",
+		"The name of the mutatingwebhookconfiguration resource in Kubernetes. Chiron will manage this resource.")
+	flags.StringVar(&opts.mutatingWebhookName, "mutating-webhook-name", "sidecar-injector.istio.io",
```

Comment:
Use plural to specify multiple webhooks. This is important because it's an CLI argument. We need to deprecate it if we change it to plural in the future.

Reply:
Thanks, will do.

Reply:
Done.